samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/util/ExtendedTimeValidator.java

```diff
@@ -0,0 +1,408 @@
+    public static final String ERROR_KEY_RETRACT_BEFORE_START         = "extended_time_retract_earlier_than_available";
+    public static final String ERROR_KEY_DUE_SAME_AS_START            = "extended_time_due_same_as_available";
+    public static final String ERROR_KEY_OPEN_WINDOW_LESS_THAN_LIMIT  = "extended_time_open_window_less_than_time_limit";
+    public static final String ERROR_KEY_USER_SUBSTRING               = "extended_time_error_user";
+    public static final String ERROR_KEY_GROUP_SUBSTRING              = "extended_time_error_group";
+
+    public static final String MSG_KEY_AND            = "extended_time_and";
+    public static final String MSG_KEY_DUP_USERS      = "extended_time_duplicate_users";
+    public static final String MSG_KEY_DUP_GROUPS     = "extended_time_duplicate_groups";
+    public static final String MSG_KEY_NAME_NOT_FOUND = "extended_time_name_not_found";
+
+    public static final String ASSESSMENT_SETTINGS_BUNDLE = "org.sakaiproject.tool.assessment.bundle.AssessmentSettingsMessages";
+
+    private static Object settingsBean;
+
+    /**
+     * Validate a single {@link ExtendedTime} entry.
+     * @param entry the {@link ExtendedTime} entry to validate
+     * @param context The {@link FacesContext}, in case validation errors are generated
+     * @param settings The SettingsBean, for quiz settings (either {@link AssessmentSettingsBean} or {@link PublishedAssessmentSettingsBean})
+     * @return true if all entities are valid; false otherwise.
+     */
+    public static boolean validateEntry( ExtendedTime entry, FacesContext context, Object settings )
+    {
+        return validateEntries( Collections.singletonList( entry ), context , settings );
+    }
+
+    /**
+     * Validate a list of {@link ExtendedTime} entries.
+     * @param entries The {@link ExtendedTime} entries to validate
+     * @param context The {@link FacesContext}, in case validation errors are generated
+     * @param settings The SettingsBean, for quiz settings (either {@link AssessmentSettingsBean} or {@link PublishedAssessmentSettingsBean})
+     * @return true if all entities are valid; false otherwise.
+     */
+    public static boolean validateEntries( List<ExtendedTime> entries, FacesContext context, Object settings )
+    {
+        boolean valid = true;
+        settingsBean = settings;
+        List<String> users = new ArrayList<>( entries.size() );
+        List<String> groups = new ArrayList<>( entries.size() );
+        for( ExtendedTime entry : entries )
+        {
+            String user = entry.getUser();
+            String group = entry.getGroup();
+            Date startDate = entry.getStartDate();
+            Date dueDate = entry.getDueDate();
+            Date retractDate = entry.getRetractDate();
+
+            if( StringUtils.isNotEmpty( user ) )
+            {
+                users.add( user );
+            }
+
+            if( StringUtils.isNotEmpty( group ) )
```

Comment:
i occassionally see instructors choose both user and group .... does it make sense to just assume that they mean one user and ignore the group selection ?

Reply:
This is the original logic. In my testing, you can still select either a user, or a group, or a user and a group.